### PR TITLE
GEOPY-715: Prototype joint inversion driver (single property)

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-715 - Prototype joint inversion driver (single property)**

### CODE REVIEW

File: geoapps/inversion/joint/single_property/driver.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+            self._inverse_problem = inverse_problem.BaseInvProblem(
+                self.data_misfit,
+                self.regularization,
+                self.optimization,
+                beta=self.params.initial_beta,
+            )
+
+        return self._inverse_problem
+
+    def validate_create_mesh(self):
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/joint/single_property/driver.py:169]: Well it also modifies the sub-drives meshes, so a lot of interaction at the high level. And that's really specific to joint. Feels like we should keep it out of the basic mesh utilities?

File: geoapps/inversion/joint/single_property/driver.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+from .params import JointSingleParams
+
+
+class JointSingleDriver(InversionDriver):
+    _params_class = JointSingleParams
+    _validations = validations
+    _drivers = None
+
+    def __init__(self, params: JointSingleParams, warmstart=True):
+        super().__init__(params, warmstart)
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/joint/single_property/driver.py:38]: As discussed in person, I will make sure they all have private attributes with setters and docstrings.

File: geoapps/inversion/joint/single_property/driver.py [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+                    driver.inversion_mesh.entity.origin["x"] - shift[0],
+                    driver.inversion_mesh.entity.origin["y"] - shift[1],
+                    driver.inversion_mesh.entity.origin["z"] - shift[2],
+                ]
+                setattr(driver.inversion_mesh, "_mesh", None)
+
+    def validate_create_models(self):
+        """Check if all models were provided."""
+        child_driver = self.drivers[0]
+        for model_type in self.models.model_types:
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/joint/single_property/driver.py:228]: Agreed.

File: geoapps/octree_creation/params.py [hunk 4]
```diff
@@ -61,7 +61,6 @@ def __init__(self, input_file=None, **kwargs):
             input_file = InputFile(
                 ui_json=ui_json,
                 validations=self.validations,
-                validation_options={"disabled": True},
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/octree_creation/params.py:64]: Correct

File: geoapps/peak_finder/params.py [hunk 5]
```diff
@@ -70,7 +70,6 @@ def __init__(self, input_file=None, **kwargs):
             input_file = InputFile(
                 ui_json=ui_json,
                 validations=self.validations,
-                validation_options={"disabled": True},
```
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/peak_finder/params.py:73]: Nice catch

File: geoapps/inversion/joint/single_property/driver.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (domfournier) [reply geoapps/inversion/joint/single_property/driver.py:228]: hmm hang on, we are actually re-assigning values to the InversionModel class...

**Review by benk-mira — APPROVED** [review]
Looks good, thanks for the updates

**Pull request merged**